core/web/src/main/java/edu/uci/ics/texera/web/resource/QueryPlanResource.java

```diff
@@ -106,6 +110,195 @@ public JsonNode executeQueryPlan(String logicalPlanJson) {
+        |                   |           //              |                   |
+        |                   |          //               |                   |
+        |___________________|         //                |___________________|
+                                     //
+                                    //
+         _______________________   //
+        |                       | //
+        |                       |//
+        | Source: Scan          |/
+        | TableName: dictionary |
+        |                       |
+        |                       |
+        |                       |
+        |_______________________|
+
+    7. After the new link generated, the Attributes field of the Keyword Search will be empty again. When
+        you try to check its drop-down list, the options are all updated to dictionary's attributes, which
+        are name and payload. The options from "plan" are all gone.
+
+    */
+    @POST
+    @Path("/autocomplete")
+    public JsonNode executeAutoQueryPlan(String logicalPlanJson) {
+        try {
+            JsonNode logicalPlanNode = new ObjectMapper().readTree(logicalPlanJson);
+            ArrayNode operators = (ArrayNode) logicalPlanNode.get(PropertyNameConstants.OPERATOR_LIST);
+            ArrayNode links = (ArrayNode) logicalPlanNode.get(PropertyNameConstants.OPERATOR_LINK_LIST);
+
+            ArrayNode validOperators = new ObjectMapper().createArrayNode();
+            ArrayNode validLinks = new ObjectMapper().createArrayNode();
+            ArrayNode linksEndWithInvalidDest = new ObjectMapper().createArrayNode();
+
+            Set<String> validOperatorsId = new HashSet<>();
+            getValidOperatorsAndLinks(operators, links, validOperators, validLinks,
+                                      linksEndWithInvalidDest, validOperatorsId);
+
+            ((ObjectNode) logicalPlanNode).putArray(PropertyNameConstants.OPERATOR_LIST).addAll(validOperators);
+            ((ObjectNode) logicalPlanNode).putArray(PropertyNameConstants.OPERATOR_LINK_LIST).addAll(validLinks);
+
+            LogicalPlan logicalPlan = new ObjectMapper().treeToValue(logicalPlanNode, LogicalPlan.class);
+            String resultID = UUID.randomUUID().toString();
+
+            // Get all input schema for valid operator with valid links
+            Map<String, List<Schema>> inputSchema = logicalPlan.retrieveAllOperatorInputSchema();
+            // Get all input schema for invalid operator with valid input operator
+            for (JsonNode linkNode: linksEndWithInvalidDest) {
+                String origin = linkNode.get(PropertyNameConstants.ORIGIN_OPERATOR_ID).textValue();
+                String dest = linkNode.get(PropertyNameConstants.DESTINATION_OPERATOR_ID).textValue();
+
+                Schema schema = logicalPlan.getOperatorOutputSchema(origin, inputSchema);
+                inputSchema.computeIfAbsent(dest, k -> new ArrayList<>()).add(schema);
```

Comment:
Same comment for `computeIfAbsent`, please check the comment above to make similar changes.